core/chains/evm/gas/fixed_price_estimator.go

```diff
@@ -43,8 +47,12 @@ type fixedPriceEstimatorBlockHistoryConfig interface {
 
 // NewFixedPriceEstimator returns a new "FixedPrice" estimator which will
 // always use the config default values for gas prices and limits
-func NewFixedPriceEstimator(cfg fixedPriceEstimatorConfig, bhCfg fixedPriceEstimatorBlockHistoryConfig, lggr logger.Logger) EvmEstimator {
-	return &fixedPriceEstimator{cfg, bhCfg, logger.Sugared(logger.Named(lggr, "FixedPriceEstimator"))}
+func NewFixedPriceEstimator(cfg fixedPriceEstimatorConfig, ethClient evmclient.Client, bhCfg fixedPriceEstimatorBlockHistoryConfig, lggr logger.Logger, chainType config.ChainType) EvmEstimator {
+	var l1Oracle rollups.L1Oracle
+	if rollups.IsRollupWithL1Support(chainType) {
```

Comment:
Curious what you think but what if we move this check into the `NewL1GasOracle` method so you don't have to have this check in all of the estimator builders? You can update `NewL1GasOracle` to return a pointer so it can return `nil` if this check doesn't pass.

Reply:
Moved into `NewL1GasOralce`, however, we are no longer using the pointer as the return value. It still works though.